libavcodec/vaapi_encode.c

```diff
@@ -1177,20 +1179,34 @@ int ff_vaapi_encode_receive_packet(AVCodecContext *avctx, AVPacket *pkt)
             return AVERROR(EAGAIN);
     }
 
-    pic = NULL;
-    err = vaapi_encode_pick_next(avctx, &pic);
-    if (err < 0)
-        return err;
-    av_assert0(pic);
+    if (av_fifo_size(ctx->encode_fifo) == 0) {
```

Comment:
Any  possible to remove this check? Or check fifo size inside while with "if (av_fifo_size(ctx->encode_fifo) == MAX_ALLOC_SIZE) break;". With your logic, if fifo size is 0, and the next pic will write to fifo, and then get it out at once, this meets the case in your commit log. If change like this, you may consider if need to enlarge fifo size in init function.

Reply:
Thanks, I try to improve code like you said.

Reply:
Done, please check.